src/test/java/io/percy/appium/AppPercyTest.java

```diff
@@ -0,0 +1,121 @@
+        capabilities.setCapability("browserstack.user", "USER_NAME");
+        capabilities.setCapability("browserstack.key", "USER_AUTH_KEY");
+        capabilities.setCapability("browserstack.appium_version", "1.20.2");
+        capabilities.setCapability("app", "APP_URL");
+        capabilities.setCapability("device", "DEVICE_NAME");
+        capabilities.setCapability("os_version", "9.0");
+        capabilities.setCapability("percy.enabled", "true");
+        capabilities.setCapability("percy.enabled", "true");
+        genericProvider = mock(GenericProvider.class);
+    }
+
+    @Test
+    public void takeScreenshotWithoutPercyEnabled() throws Exception {
+      when(androidDriver.getSessionId()).thenReturn(new SessionId("123"));
+      
+      try {
+        lenient(). when(androidDriver.getRemoteAddress()).thenReturn(new URL("https://hub.browserstack.com/wd/hub"));
+      } catch (MalformedURLException e) {
+        throw new RuntimeException(e);
+      }
+
+      percy = spy(new AppPercy(androidDriver));
+      percy.setGenericProvider(genericProvider);
+    
+      JSONObject res = percy.screenshot("Test");
+      assertEquals(res, null);
+      verify(genericProvider, never()).screenshot(eq("Test"), eq(null));
+    }
+
+    @Test
+    public void takeScreenshotWithoutOptions() throws Exception {
+        // Percy Options to Enable Percy
+        capabilities.setCapability("percy.enabled", "true");
+
+      when(androidDriver.getSessionId()).thenReturn(new SessionId("123"));
+      when(androidDriver.getCapabilities()).thenReturn(capabilities);
+
+      try {
+          lenient(). when(androidDriver.getRemoteAddress()).thenReturn(new URL("https://hub.browserstack.com/wd/hub"));
+      } catch (MalformedURLException e) {
+          throw new RuntimeException(e);
+      }
+
+      ArgumentCaptor<ScreenshotOptions> captureOptions = ArgumentCaptor.forClass(ScreenshotOptions.class);
+      percy = spy(new AppPercy(androidDriver));
+      percy.setGenericProvider(genericProvider);
+      percy.setPercyEnabled();
+      when(genericProvider.screenshot(any(), any())).thenReturn(null);
+      JSONObject res = percy.screenshot("Test");
+      assertEquals(res, null);
+      verify(genericProvider).screenshot(eq("Test"), captureOptions.capture());
+      assertNotNull(captureOptions.getValue());
+      assertTrue(captureOptions.getValue() instanceof ScreenshotOptions);
+    }
+
+    @Test
+    public void takeScreenshotWithSyncOption() throws Exception{
+
+      when(androidDriver.getSessionId()).thenReturn(new SessionId("123"));
+      when(androidDriver.getCapabilities()).thenReturn(capabilities);
+      
+      try {
+        lenient(). when(androidDriver.getRemoteAddress()).thenReturn(new URL("https://hub.browserstack.com/wd/hub"));
+      } catch (MalformedURLException e) {
+        throw new RuntimeException(e);
+      }
+      
+      percy = spy(new AppPercy(androidDriver));
+      percy.setGenericProvider(genericProvider);
```

Comment:
No need to set generic provider right? As we are mocking getRemoteAddressCall.

Reply:
We need to add to mock response `provider.screenshot` command.

Reply:
Than above that function write a comment that this is only for test.

Reply:
ack.